src/utils.ts

```diff
@@ -198,6 +200,21 @@ export const isSafari = (): boolean => {
   ); // safari
 };
 
+/**
+ * Regular expression to validate GGUF file paths/URLs
+ * Matches paths ending with .gguf and optional query parameters
+ */
+export const GGUF_FILE_REGEX = /^.*\.gguf(?:\?.*)?$/;
+
+/**
+ * Validates if a given string is a valid GGUF file path/URL
+ * @param path The file path or URL to validate
+ * @returns true if the path is a valid GGUF file path/URL
+ */
+export const isValidGgufFile = (path: string): boolean => {
+  return GGUF_FILE_REGEX.test(path);
```

Comment:
I think we only need to check if it contains `.gguf`, so it can be more simple:

```suggestion
  return path.includes('.gguf');
```

Reply:
ignore this, your test case for catching `model.gguf.bin` seems to be a good idea, I decide to keep this regex

Reply:
No worries!

How about case sensitivity?
Do you think it should allow `.GGUF`, `.Gguf`, `.GGuF` and similar cases?
Currently, it will only accept all lower-cased.

Reply:
lowercase is ok, I think no one will gonna use uppercase